sync/sync.go

```diff
@@ -84,22 +84,30 @@ func (s *Synchronizer) fetcherTask(ctx context.Context, height uint64, verifiers
 func (s *Synchronizer) verifierTask(ctx context.Context, block *core.Block, stateUpdate *core.StateUpdate,
 	declaredClasses map[felt.Felt]core.Class, resetStreams context.CancelFunc,
 ) stream.Callback {
-	err := s.Blockchain.SanityCheckNewHeight(block, stateUpdate)
+	errs := s.Blockchain.SanityCheckNewHeight(block, stateUpdate, declaredClasses)
 	return func() {
 		select {
 		case <-ctx.Done():
 			return
 		default:
-			if err != nil {
-				if errors.As(err, new(core.CantVerifyTransactionHashError)) {
-					for ; err != nil; err = errors.Unwrap(err) {
-						s.log.Debugw("Sanity checks failed", "number", block.Number, "hash",
-							block.Hash.ShortString(), "error", err.Error())
+			for _, err := range errs {
+				if err != nil {
+					var cErr *core.ErrCantVerifyClassHash
+					if errors.As(err, &cErr) {
+						for ; err != nil; err = errors.Unwrap(err) {
+							s.log.Debugw("Sanity checks failed", "number", block.Number, "hash",
+								block.Hash.ShortString(), "error", cErr.Error())
+						}
+					} else if errors.As(err, new(core.CantVerifyTransactionHashError)) {
+						for ; err != nil; err = errors.Unwrap(err) {
+							s.log.Debugw("Sanity checks failed", "number", block.Number, "hash",
+								block.Hash.ShortString(), "error", err.Error())
+						}
+					} else {
+						s.log.Warnw("Sanity checks failed", "number", block.Number, "hash", block.Hash.ShortString())
+						resetStreams()
+						return
```

Comment:
Since we ignore the error which fails class hash and transaction verification. Do you think it would be best to log the errors in `SanityCheckNewHeight` and return nil if class or transaction verification fails? 

This would mean that we would have to write separate tests for transaction verification as you have done for class.

Reply:
Sounds good to me.

Reply:
Done